### PR TITLE
Remove use of test priorities and dependsOnMethods from Prometheus tests

### LINKED ISSUE #6060: Remove use of test priorities and dependsOnMethods from Prometheus tests
Remove things like `@Test(dependsOnMethods ...)` or `@Test(priority ...)`

### PR DESCRIPTION
**Pull request by m2yi**
Fixes #6060

### CODE REVIEW

**Comment by findepi** [comment]
Thanks @m2yi for your PR!

Changes look good.
I've added a temporary commit to exercise the code more than typically done for a PR, so make sure no test flakiness gets introduced.
I hope you don't mind me pushing to your PR branch.

**Comment by m2yi** [comment]
no problem, @findepi! 

For now, seems like all checks passed, so pls let me know if I need to update something from my side. Also, I've already signed CLA and applied comments from @cla-bot, so could you pls advice on how can I get signed this PR?

**Comment by findepi** [comment]
Thanks @m2yi for confirming build results!
For CLA -- we need @martint 's help. If you already signed it, you do not need to do anything more.

**Comment by m2yi** [comment]
Hi @martint, could you pls help to check whether my CLA was signed?

**Comment by findepi** [comment]
@cla-bot check

**Comment by findepi** [comment]
@m2yi your license has been processed, but the CLA is not cleared for the PR yet.
This is because the commits are not associated with your email address.
Please refer to cla-bot's advice on how to fix that.

If you're rebasing, you can remove my TMP commit, it's no longer needed.
Thanks

**Comment by m2yi** [comment]
@findepi thanks for help, seems like CLA signed now and all checks passed
pls, let me know if I need to update something else

File: plugin/trino-prometheus/src/test/java/io/trino/plugin/prometheus/TestPrometheusIntegrationMetrics.java [hunk 1]
```diff
@@ -100,7 +101,7 @@ public void testGetColumnMetadata()
         // directly.
     }
 
-    @Test(expectedExceptions = TrinoException.class, dependsOnMethods = "testRetrieveUpValue")
+    @Test(expectedExceptions = TrinoException.class)
```
Comment (findepi) [inline plugin/trino-prometheus/src/test/java/io/trino/plugin/prometheus/TestPrometheusIntegrationMetrics.java:104]: as a followup we should remove `expectedExceptions` use and replace it with `assertThatThrownBy`

**Comment by findepi** [comment]
Thank you!

File: plugin/trino-prometheus/src/test/java/io/trino/plugin/prometheus/TestPrometheusIntegrationStatus.java [hunk 2]
```diff
@@ -97,7 +97,7 @@ public void testConfirmMetricAvailableAndCheckUp()
         assertEquals(row.getField(0).toString(), "{instance=localhost:9090, __name__=up, job=prometheus}");
     }
 
-    @Test(priority = 2, dependsOnMethods = "testConfirmMetricAvailableAndCheckUp")
```
Comment (ebyhr) [inline plugin/trino-prometheus/src/test/java/io/trino/plugin/prometheus/TestPrometheusIntegrationStatus.java:100]: It seems `testPushDown()` fails if  we don't run `testConfirmMetricAvailableAndCheckUp()` beforehand.
```sh
java.lang.RuntimeException: line 1:22: Table 'prometheus.default.up' does not exist

	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:122)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:505)
	at io.trino.testing.AbstractTestQueryFramework.computeActual(AbstractTestQueryFramework.java:139)
	at io.trino.testing.AbstractTestQueryFramework.computeActual(AbstractTestQueryFramework.java:134)
	at io.trino.plugin.prometheus.TestPrometheusIntegrationStatus.testPushDown(TestPrometheusIntegrationStatus.java:111)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
	at org.testng.TestNG.runSuites(TestNG.java:1133)
	at org.testng.TestNG.run(TestNG.java:1104)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
	Suppressed: java.lang.Exception: SQL: SELECT now(), * FROM prometheus.default.up WHERE timestamp > (NOW() - INTERVAL '15' SECOND)
		at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:508)
		... 27 more
Caused by: io.trino.spi.TrinoException: line 1:22: Table 'prometheus.default.up' does not exist
	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:48)
	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:43)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:1520)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:404)
	at io.trino.sql.tree.Table.accept(Table.java:60)
	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:421)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.analyzeFrom(StatementAnalyzer.java:3405)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:2144)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:404)
	at io.trino.sql.tree.QuerySpecification.accept(QuerySpecification.java:155)
	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:421)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:429)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:1347)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:404)
	at io.trino.sql.tree.Query.accept(Query.java:107)
	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:421)
	at io.trino.sql.analyzer.StatementAnalyzer.analyze(StatementAnalyzer.java:384)
	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:79)
	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:71)
	at io.trino.execution.SqlQueryExecution.analyze(SqlQueryExecution.java:254)
	at io.trino.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:179)
	at io.trino.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:776)
	at io.trino.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:132)
	at io.trino.$gen.Trino_testversion____20220125_060823_3.call(Unknown Source)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:74)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Pull request merged**